app/src/main/kotlin/at/bitfire/davdroid/servicedetection/DavResourceFinder.kt

```diff
@@ -56,12 +59,18 @@ import java.util.logging.Logger
  * @param baseURI        user-given base URI (either mailto: URI or http(s):// URL)
  * @param credentials    optional login credentials (username/password, client certificate, OAuth state)
  */
-class DavResourceFinder(
-    val context: Context,
-    private val baseURI: URI,
-    private val credentials: Credentials? = null
+class DavResourceFinder @AssistedInject constructor(
+    @Assisted private val baseURI: URI,
+    @Assisted private val credentials: Credentials? = null,
+    @ApplicationContext val context: Context,
+    private val dnsRecordResolver: DnsRecordResolver
 ): AutoCloseable {
 
+    @AssistedFactory
+    interface Factory {
+        fun resourceFinder(baseURI: URI, credentials: Credentials?): DavResourceFinder
```

Comment:
I am not 100% sure, but we usually simply call this method "create" no? Because it's clear what is being created from the factory name already.

Reply:
Given that there must be exactly one method in a `@AssistedFactory`, there shouldn't be any confusion when the method is always named `create`. Also in https://dagger.dev/dev-guide/assisted-injection.html they name it `create` too. So let's stick with this convention.